utility-types/Exercise3/solution/Solution.ts

```diff
@@ -0,0 +1,65 @@
+type Person =
+  | {
+      role: "developer"
+      email: string
+      id: number
+      firstName: string
+      lastName: string
+      team: "React" | "Angular" | "backend"
+    }
+  | {
+      role: "user"
+      email: string
+      id: number
+      firstName: string
+      lastName: string
+      isVerified: boolean
+    }
+
+type UserMap = Record<Person["email"], Person>
+
+const people: Person[] = [
+    {
+      role: "developer",
+      email: "email@developer.com",
+      firstName: "Dev",
+      lastName: "Eloper",
+      team: "React",
+      id: 1,
+    },
+    {
+      role: "developer",
+      email: "jane@developer.com",
+      firstName: "Dev",
+      lastName: "Eloper",
```

Comment:
Name should probably be different

```suggestion
      firstName: "Jay",
      lastName: "Ane",
```